thorlcr/activities/thactivityutil.cpp

```diff
@@ -234,7 +245,15 @@ class CRowStreamLookAhead : public CSimpleInterfaceOf<IStartableEngineRowStream>
 // IEngineRowStream
     virtual const void *nextRow() override
     {
-        OwnedConstThorRow row = smartbuf->nextRow();
+        OwnedConstThorRow row;
+        {
+            // smartbuf->nextRow should return immediately if a row is available.
+            // smartbuf->nextRow will take time if blocked, so record time taken as blocked time.
+            // N.b. smartbuf->next may take a trivial amount of time if row is available but
+            // for the purposes of stats this will still be considered blocked.
+            BlockedActivityTimer timer(activity.getActivityTimerAccumulator(), activity.queryTimeActivities());
```

Comment:
so lookahead time accumulates the time taken reading from the input, but not time it takes pushing it to the writer (won't available until it is).
But this will factor that time in, the time reader calling this spends waiting for the rows to be available (blocked), which will include the time the lookahead takes adding the row to the writer, if this is blocked on it.

I think this is correct, just talking it through :) 

@ghalliday - can you look at this also?

Reply:
I think that is reasonable.  It means time spent adding rows will not be recorded anywhere, but that is probably ok since they are likely to be overlapping with other operations.